torch/csrc/jit/tracer.cpp

```diff
@@ -102,6 +102,13 @@ Value* getValueTrace(const IValue& var) {
     }
 
     // Didn't find it. Bake in a constant
+    if (ten.requires_grad()) {
+      std::ostringstream oss;
+      oss << "Cannot insert a Tensor that requires grad as a constant. "
+          << "Consider making it a parameter or input, or detaching the gradient\n";
+      throw std::runtime_error(oss.str());
```

Comment:
I had to remove the print to tensor because printing calls .contiguous which re-enters the tracer, causing a stack trace failure. That should be fixed but should be in a follow up PR